lib/shopify_api/session.rb

```diff
@@ -67,7 +67,7 @@ def encoded_params_for_signature(params)
       end
     end
 
-    def initialize(url, token = nil, api_version = :no_version, extra = {})
+    def initialize(url:, token:, api_version:, extra: {})
```

Comment:
Should we take this opportunity to also rename `url` to `domain`, to be consistent with `temp`, but also to choose a name that honours the URL/URI specs? https://doepud.co.uk/blog/anatomy-of-a-url

Another name to consider might be `shop_domain`?

Reply:
I agree I put it up like this so we could make a call on what this should look like.

Reply:
alternative would be to call it `site` since that is the name used for this value in ActiveResource.

Reply:
Depends on whether that leaks any of the abstraction?

Reply:
review everything closer.  `domain` is the best name since we will produce a full url for `site` already.